tools/cherry-pick/bin/run.js

```diff
@@ -0,0 +1,580 @@
+ *
+ * @param string msg The message for the commit.
+ */
+function commitChanges( msg ) {
+	const spinner = ora( {
+		spinner: 'bouncingBar',
+		color: 'green'
+	} );
+
+	let output = [];
+
+	return new Promise( ( resolve, reject ) => {
+		const response = spawn( 'git', [ 'commit', '--no-verify', '-am', msg ] );
+
+		response.stdout.on( 'data', ( data ) => {
+			output.push( `${data}` );
+		} );
+
+		response.stderr.on( 'data', ( data ) => {
+			output.push( `${data}` );
+		} );
+
+		response.on( 'close', ( code ) => {
+			if ( `${code}` == 0 ) {
+				spinner.succeed( `Commited changes.` );
+				resolve();
+			}
+
+			reject( `error: ${output.toString().replace( ',', "\n" )}` );
+		} );
+	} ).catch( err => {
+		spinner.fail( 'Fail to commit changes.' );
+		throw err;
+	} );
+}
+
+/**
+ * Push the branch up to GitHub.
+ *
+ * @param string branch The branch to push to GitHub.
+ */
+function pushBranch( branch ) {
+	const spinner = ora( {
+		spinner: 'bouncingBar',
+		color: 'green'
+	} );
+
+	spinner.start( `Pushing ${branch} to GitHub...` );
+
+	let output = [];
+
+	return new Promise( ( resolve, reject ) => {
+		const response = spawn( 'git', [ 'push', 'origin', branch ] );
+
+		response.stdout.on( 'data', ( data ) => {
+			output.push( `${data}` );
+		} );
+
+		response.stderr.on( 'data', ( data ) => {
+			output.push( `${data}` );
+		} );
+
+		response.on( 'close', ( code ) => {
+			if ( `${code}` == 0 ) {
+				spinner.succeed();
+				resolve();
+			}
+
+			reject( `error: ${output.toString().replace( ',', "\n" )}` );
+		} );
+	} ).catch( err => {
+		spinner.fail( `Fail to push ${branch} up.` );
+		throw err;
+	} );
+}
+
+/**
+ * Create pull request on GitHub.
+ *
+ * @param string title The title of the PR.
+ * @param string body The body content of the PR.
+ * @param string head The head of the branch to use.
+ * @param string base The base branch to targe against.
+ */
+async function createPR( title, body, head, base ) {
+	const spinner = ora( {
+		spinner: 'bouncingBar',
+		color: 'green'
+	} );
+
+	spinner.start( `Creating pull request for ${head} on GitHub...` );
+
+	const response = await fetch( 'https://api.github.com/repos/woocommerce/woocommerce/pulls', {
+		method: 'POST',
+		headers: {
+			'Accept': 'application/vnd.github.v3+json',
+			'Authorization': `token ${githubToken}`,
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify( {
+			"title": title,
+			"body": body,
+			"head": head,
+			"base": base
+		} )
+	} );
+
+	if ( response.status !== 201 ) {
+		throw 'Fail to create PR on GitHub.';
+	}
+
+	return await response.json().then( data => {
+		spinner.succeed();
+		return {
+			"url": data.url,
+			"number": data.number
+		}
+	} );
+}
+
+( async function() {
+	try {
+		// Creates a temp directory to work with.
+		await createDir( path.join( os.tmpdir(), 'woo-cherry-pick-') );
+
+		// Gets the commits from GitHub based on the PRs passed in.
+		await getCommitFromPrs( prsArr );
+
+		chdir( tempWooDir );
```

Comment:
Out of curiosity, what are the benefits of this approach vs using `cwd` in all of the `spawn` calls? This feels like it might be prone to confusion in the future if someone makes a change without realizing the commands are all directory-sensitive.

Reply:
No benefits or disadvantage I would say. I just chose the path of least resistance at the time not having to pass CWD to all spawns and instead call `chdir` twice.